plugin/trino-faker/src/main/java/io/trino/plugin/faker/FakerConfig.java

```diff
@@ -26,6 +26,7 @@ public class FakerConfig
     private double nullProbability = 0.5;
     private long defaultLimit = 1000L;
     private Locale locale = Locale.ENGLISH;
+    private long maxDictionarySize = 1000L;
```

Comment:
Why is low cardinality in particular a special case ? Ideally we know the NDV and should be able to adjust the fake data generation to that NDV, doesn't have to exact same NDV.
If the range of data is small I think you'll get small cardinality automatically.

Reply:
This is exactly what's happening here - adjusting the NDV. We can't adjust it with range alone, as it'll skew the stats of the generated data.

I added this threshold as a safeguard against having too many values in a column property, I don't know what are the practical limits there. Also, generating this many unique values can get expensive, too.

Reply:
If even you don't know the limits, how can the user possibly know ?
I suggest defining a constant in the code and doing this without exposing configs to the user. I can't see why the user would want to tune this.
Also, the column property size limit is an artifical constraint coming from how this functionality is implemented (going through ALLOWED_VALUES_PROPERTY). We aren't getting the unique values from the source connnector, so faker connector shouldn't need anything more than the unique values count to generate the random dictionary itself.

Reply:
I made it configurable, because I don't know what's a good default value. I choose 1000 as a wild guess, but it's also pretty low. I understand we want to have as few knobs as possible, but I'm really not confident in choosing the default here. Is 10000 still ok? That's going to look terrible in the output of `SHOW CREATE TABLE`. I wanted to leave it up to users to deal with that.

Reply:
You are choosing the default anyway, most ppl aren't going to find the config or really understand what it does. I prefer to avoid putting this in user hands. I suggest using 1000 or something small as the constant in the code or find a way to re-implement this in a way that doesn't rely on ALLOWED_VALUES_PROPERTY, then the properties limit doesn't matter

Reply:
Should we allow users to disable detecting low cardinality columns, with a boolean config and/or schema/table properties?

Reply:
Just a boolean config should be fine as a kill switch if you foresee something going wrong with it. I wouldn't bother at this early stage of this connector where its not a critical functionality yet.

Reply:
All done! Thanks for your patience with me, I agree this looks better now. I also found and fixed a bug in how the number of generated allowed values was picked.